### PR TITLE
Vec3 * Quat

### PR DESCRIPTION
**Pull request by Pryme8**
I don't think we have a method for this yet.

If we already do this can be skipped if not then here we go.

### CODE REVIEW

File: src/Maths/math.vector.ts [hunk 1]
```diff
@@ -1090,6 +1090,60 @@ export class Vector3 {
...
+        num10 = quat.w * num,
+        num11 = quat.w * num2,
+        num12 = quat.w * num3;
+
+        let x = (((1 - (num5 + num6)) * this.x) + ((num7 - num12) * this.y)) + ((num8 + num11) * this.z),
+        y = (((num7 + num12) * this.x) + ((1 - (num4 + num6)) * this.y)) + ((num9 - num10) * this.z),
+        z = (((num8 - num11) * this.x) + ((num9 + num10) * this.y)) + ((1 - (num4 + num5)) * this.z);
+
+        this.set(x, y, z);
+
```
Comment (Hersir88) [inline src/Maths/math.vector.ts:1117]: this multiplyByQuatToRef(quat, this), no need to duplicate code

Comment (Hersir88) [reply src/Maths/math.vector.ts:1117]: @deltakosh would know better but would assume that this should copy vector and return that, to follow other methods in the, file, as there are 3 types of them standard returns fresh copy, `*inPlace` modifies current vector and `*ToRef` passed one

**Comment by RaananW** [comment]
Is this the same as `Vector3.rotateByQuaternionToRef` ?

**Comment by deltakosh** [comment]
I agree this is duplicate code

**Comment by Pryme8** [comment]
> Is this the same as `Vector3.rotateByQuaternionToRef` ?

Now that I look at it yes it is.  My bad, for some reason last night I thought they were different then this morning I was gonna close it anyways because I kind of realized that as well.

*sigh* the things we think of when we are tired and just getting stuff done.

Thanks for taking a look though!

**Pull request closed without merging**